rlclientlib/logger/async_batcher.h

```diff
@@ -52,9 +52,13 @@ namespace reinforcement_learning { namespace logger {
   private:
     int fill_buffer(std::shared_ptr<utility::data_buffer>& retbuffer,
       size_t& remaining,
-      api_status* status);
+      api_status* status,
+      unsigned int* original_count);
 
     void flush(); //flush all batches
+    void increment_counter();
+    unsigned int reset_counter_get_count();
+    std::mutex _counter_mutex;
```

Comment:
super nit: maybe move the mutex to be with the other member variables below to follow the structure of the file